includes/Core/Permissions/Permissions.php

```diff
@@ -241,26 +241,48 @@ function( array $allcaps ) {
 		);
 	}
 
+	/**
+	 * Get dashboard sharing meta permissions for current user.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array List meta capabilities as keys and current user permission as value.
+	 */
+	public function get_dashboard_sharing_meta_permissions() {
+		if ( ! Feature_Flags::enabled( 'dashboardSharing' ) ) {
+			return array();
+		}
+
+		$dashboard_sharing_meta_capabilities = self::get_dashboard_sharing_meta_capabilities();
+
+		$shareable_modules = array_keys( $this->modules->get_shareable_modules() );
+
+		$dashboard_sharing_meta_permissions = array();
+		foreach ( $dashboard_sharing_meta_capabilities as $cap ) {
+			foreach ( $shareable_modules as $module ) {
+				$dashboard_sharing_meta_permissions[ "{$cap}::['{$module}']" ] = current_user_can( $cap, $module );
```

Comment:
@hussain-t Why is this not using `wp_json_encode( $module )` as outlined in the IB? If the problem is that it produces double quotes, we could also wrap the overall string into single quotes to avoid the need for extra escaping here.

"Manual JSON-encoding" like here is risky and could easily break. While with our limited use-case here it's highly unlikely, I suggest we change this to make it less error-prone. Imagine `$module` contained a single quote right now - of course it shouldn't, but that would cause an error here, and by using `wp_json_encode` that can be avoided.

Reply:
cc @jimmymadon @tofumatt

Reply:
Hi @felixarntz, this is a good shout, fortunately I can confirm it's been addressed in a subsequent PR, as can be seen here: https://github.com/google/site-kit-wp/pull/4907/files

Reply:
@techanvil Ah, thanks for clarifying, I hadn't seen this one. In that case, all good! 👍